parsec/core/fs/storage/chunk_storage.py

```diff
@@ -253,18 +266,26 @@ async def cleanup(self, cursor: Cursor | None = None) -> None:
 
             # No clean up is needed
             if extra_blocks <= 0:
-                return
+                return []
 
             # Remove the extra block plus 10 % of the cache size, i.e about 100 blocks
             limit = extra_blocks + self.block_limit // 10
+
             # Use a thread as executing a statement that modifies the content of the database might,
             # in some case, block for several hundreds of milliseconds
-            await self.localdb.run_in_thread(
-                cursor.execute,
-                """
-                DELETE FROM chunks WHERE chunk_id IN (
-                    SELECT chunk_id FROM chunks ORDER BY accessed_on ASC LIMIT ?
+            def _thread_target(cursor: Cursor) -> list[ChunkID]:
+                cursor.execute(
+                    """
+                    DELETE FROM chunks WHERE chunk_id IN (
+                        SELECT chunk_id FROM chunks ORDER BY accessed_on ASC LIMIT ?
+                    ) RETURNING chunk_id
```

Comment:
Add a comment to indicate this cool returning is available when re-writing the code in rust (since we ship a more recent version of sqlite)